### PR TITLE
LoopVectorize/test: add missing CHECK lines, cleanup intrinsic.ll

### PR DESCRIPTION
**Pull request by artagnon**
Clean up intrinsic.ll by removing extraneous attributes and target datalayout, fix a bug in the copysign_f64 test, and add missing CHECK lines.

### CODE REVIEW

**Comment by fhahn** [comment]
Not sure if it is worth auto-generating the checks here; the file mainly tests that we can widen various intrinsics and auto-generating checks here will mean that the test will need updating any time in the future when anything unrelated to intrinsic handling cause changes, adding additional churn.

It also increases the size of the test quite a bit, so if we go with auto-generating it might make sense to split the file up a bit.

**Comment by artagnon** [comment]
Makes sense, although I've had several bad experiences with hand-crafted CHECK lines. Please see updated patch.

File: llvm/test/Transforms/LoopVectorize/intrinsic.ll [hunk 1]
```diff
@@ -1058,18 +1101,19 @@ for.end:                                          ; preds = %for.body, %entry
   ret void
 }
 
+define void @fabs_libm(ptr nocapture %x) nounwind {
 ; CHECK: fabs_libm
-; CHECK:  call <4 x float> @llvm.fabs.v4f32
+; CHECK: llvm.fabs.f32
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/intrinsic.ll:1106]: This is a change in the test case? Are the attributes needed here?

File: llvm/test/Transforms/LoopVectorize/intrinsic.ll [hunk 2]
```diff
@@ -1510,12 +1568,13 @@ for.end:                                          ; preds = %for.body, %entry
...
+declare float @llvm.maximum.f32(float, float)
 
-;CHECK-LABEL: @maximum_f32(
-;CHECK: llvm.maximum.v4f32
-;CHECK: ret void
-define void @maximum_f32(i32 %n, ptr noalias %y, ptr noalias %x, ptr noalias %z) nounwind uwtable {
+define void @maximum_f32(i32 %n, ptr %y, ptr %x, ptr %z) {
+; CHECK-LABEL: @maximum_f32(
+; CHECK: llvm.maximum.v4f32
+; CHECK: ret void
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/intrinsic.ll:1576]: wrapping looks strange here.

File: llvm/test/Transforms/LoopVectorize/intrinsic.ll [hunk 3]
```diff
@@ -493,9 +513,13 @@ for.end:                                          ; preds = %for.body, %entry
   ret void
 }
 
-declare float @llvm.copysign.f32(float, float) nounwind readnone
+declare float @llvm.copysign.f32(float, float)
 
-define void @copysign_f64(i32 %n, ptr noalias %y, ptr noalias %x, ptr noalias %z) nounwind uwtable {
+define void @copysign_f64(i32 %n, ptr %y, ptr %x, ptr %z) {
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/intrinsic.ll:518]: wrapping here looks odd

File: llvm/test/Transforms/LoopVectorize/intrinsic.ll [hunk 4]
```diff
@@ -465,12 +484,13 @@ for.end:                                          ; preds = %for.body, %entry
...
+declare double @llvm.fabs(double)
 
-;CHECK-LABEL: @copysign_f32(
-;CHECK: llvm.copysign.v4f32
-;CHECK: ret void
-define void @copysign_f32(i32 %n, ptr noalias %y, ptr noalias %x, ptr noalias %z) nounwind uwtable {
+define void @copysign_f32(i32 %n, ptr %y, ptr %x, ptr %z) {
+; CHECK-LABEL: @copysign_f32(
+; CHECK: llvm.copysign.v4f32
+; CHECK: ret void
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/intrinsic.ll:492]: wrapping here looks off

File: llvm/test/Transforms/LoopVectorize/intrinsic.ll [hunk 5]
```diff
@@ -1,11 +1,10 @@
 ; RUN: opt < %s -passes=loop-vectorize,dce,instcombine -force-vector-interleave=1 -force-vector-width=4 -S | FileCheck %s
 
-target datalayout = "e-p:64:64:64-i1:8:8-i8:8:8-i16:16:16-i32:32:32-i64:64:64-f32:32:32-f64:64:64-v64:64:64-v128:128:128-a0:0:64-s0:64:64-f80:128:128-n8:16:32:64-S128"
-
-;CHECK-LABEL: @sqrt_f32(
-;CHECK: llvm.sqrt.v4f32
-;CHECK: ret void
-define void @sqrt_f32(i32 %n, ptr noalias %y, ptr noalias %x) nounwind uwtable {
+define void @sqrt_f32(i32 %n, ptr %y, ptr %x) {
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/intrinsic.ll:3]: wrapping here looks off

**Review by fhahn — COMMENTED** [review]
Thanks for the updates! A few more comments inline

**Comment by fhahn** [comment]
> I agree that it looks a bit strange on the GitHub UI, but it looks fine in my editor.

If it is fine in the editor that's good, sounds like just a glitch in the Github UI

File: llvm/test/Transforms/LoopVectorize/intrinsic.ll (hunk 1 above)
Comment (artagnon) [reply llvm/test/Transforms/LoopVectorize/intrinsic.ll]: The attributes were indeed necessary. Thanks for catching! Fixed now.

**Comment by artagnon** [comment]
Gentle ping.

File: llvm/test/Transforms/LoopVectorize/intrinsic.ll [hunk 6]
```diff
@@ -506,7 +530,7 @@ for.body:                                         ; preds = %entry, %for.body
   %0 = load double, ptr %arrayidx, align 8
   %arrayidx1 = getelementptr inbounds double, ptr %z, i64 %indvars.iv
   %1 = load double, ptr %arrayidx, align 8
```
Comment (nikic) [inline llvm/test/Transforms/LoopVectorize/intrinsic.ll:532]:
```suggestion
  %1 = load double, ptr %arrayidx1, align 8
```
I think this was the intention. Currently the copysign call gets dropped by InstCombine.

File: llvm/test/Transforms/LoopVectorize/intrinsic.ll [hunk 7]
```diff
@@ -570,12 +596,12 @@ for.end:                                          ; preds = %for.body, %entry
   ret void
 }
 
-declare double @llvm.floor.f64(double) nounwind readnone
+declare double @llvm.floor.f64(double)
 
 ;CHECK-LABEL: @ceil_f32(
 ;CHECK: llvm.ceil.v4f32
 ;CHECK: ret void
```
Comment (nikic) [inline llvm/test/Transforms/LoopVectorize/intrinsic.ll:603]: Leftover old formatting?

File: llvm/test/Transforms/LoopVectorize/intrinsic.ll (hunk 6 above)
Comment (artagnon) [reply llvm/test/Transforms/LoopVectorize/intrinsic.ll]: Excellent catch! Indeed, llvm.copysign.v4f64 should be matched by the fixed test.

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**